app/assets/javascripts/osem-datepickers.js

```diff
@@ -1,33 +1,107 @@
+// get current_date
+var today = new Date().toISOString().slice(0, 10);
 $(function () {
-  $("#registration-arrival-datepicker, #registration-departure-datepicker").datetimepicker({
+  $("#registration-arrival-datepicker").datetimepicker({
       pickTime: true,
       useCurrent: false,
       minuteStepping: 15,
       sideBySide: true,
-      format: "YYYY-MM-DD HH:mm"
+      format: "YYYY-MM-DD HH:mm",
+      // current_date <= arrival_date <= end_date
+      maxDate : $("#registration-arrival-datepicker").attr('end_date'),
+      minDate : today,
+      defaultDate :   $("#registration-arrival-datepicker").attr('start_date'),
+  });
+  $("#registration-departure-datepicker").datetimepicker({
+      pickTime: true,
+      useCurrent: false,
+      minuteStepping: 15,
+      sideBySide: true,
+      format: "YYYY-MM-DD HH:mm",
+      // departure_date > start_date
+      minDate : $("#registration-arrival-datepicker").attr('start_date'),
+      defaultDate :   $("#registration-arrival-datepicker").attr('end_date'),
   });
 
   $("#registration-arrival-datepicker").on("dp.change",function (e) {
-      console.log (e.date)
-      $('#registration-departure-datepicker').data("DateTimePicker").setDate(e.date);
-      $('#registration-departure-datepicker').data("DateTimePicker").setMinDate(e.date);
-  });
-  $("#conference-end-datepicker").on("dp.change",function (e) {
-      $('#registration-arrival-datepicker').data("DateTimePicker").setMaxDate(e.date);
+    //   $('#registration-departure-datepicker').data("DateTimePicker").setDate(e.date);
+    // console.log(new Date(e.date).getTime() ,new Date($("#registration-arrival-datepicker").attr('start_date')).getTime())
+
+    // departure_date > start_date,arrival_date
+    if ((new Date(e.date).getTime()) > (new Date($("#registration-arrival-datepicker").attr('start_date')).getTime())){
+            // console.log (e.date, "arrival")
+          $('#registration-departure-datepicker').data("DateTimePicker").setMinDate(e.date);
+    }
+    else{
+        //   console.log ($("#registration-arrival-datepicker").attr('start_date'),"start_date")
```

Comment:
```Js
+    //   $('#registration-departure-datepicker').data("DateTimePicker").setDate(e.date);
+    // console.log(new Date(e.date).getTime() ,new Date($("#registration-arrival-datepicker").attr('start_date')).getTime())
+            // console.log (e.date, "arrival")
+        //   console.log ($("#registration-arrival-datepicker").attr('start_date'),"start_date")
+       //   defaultDate : today,
+    //   defaultDate : $("#registration-period-start-datepicker").attr('end_date'),
+    //   $('#conference-end-datepicker').data("DateTimePicker").setDate(e.date);
+    //   $('#registration-period-end-datepicker').data("DateTimePicker").setDate(e.date);
```

Don't comment lines, if you don't want the functionality of some line just remove it.